approaching_emergency_vehicle_plugin/src/approaching_emergency_vehicle_plugin_node.cpp

```diff
@@ -508,12 +648,611 @@ namespace approaching_emergency_vehicle_plugin
+          if(!wm_->getMapRoutingGraph()->following(current_lanelet, false).empty())
+          {
+            current_lanelet = wm_->getMapRoutingGraph()->following(current_lanelet, false).front();
+          }
+          else
+          {
+            RCLCPP_WARN_STREAM(rclcpp::get_logger(logger_name), "A following lanelet in the current lane could not be found; returning empty plan");
+            resp->new_plan.maneuvers = {};
+            return;
+          }
+
+          // Check whether the next lanelet is on the route
+          if(!wm_->getRoute()->contains(current_lanelet)){
+            RCLCPP_WARN_STREAM(rclcpp::get_logger(logger_name), "The next lanelet in the current lane is not on the route; returning empty plan");
+            resp->new_plan.maneuvers = {};
+            return;
+          }
+
+          // Update the maneuver parameters for the next maneuver in the plan
+          downtrack_progress = wm_->routeTrackPos(current_lanelet.centerline2d().front()).downtrack;
+          current_lanelet_ending_downtrack = wm_->routeTrackPos(current_lanelet.centerline2d().back()).downtrack;
+          speed_progress = getManeuverEndSpeed(resp->new_plan.maneuvers.back());
+          target_speed = getLaneletSpeedLimit(current_lanelet);
+          time_progress += getManeuverDuration(resp->new_plan.maneuvers.back(), epsilon_);
+
+          RCLCPP_DEBUG_STREAM(rclcpp::get_logger(logger_name), "Next maneuver starting downtrack is " << downtrack_progress << ", end of plan is at " << maneuver_plan_ending_downtrack);
+        }
+      }
+    }
+    else{
+      bool first_maneuver = true;
+      while(downtrack_progress < maneuver_plan_ending_downtrack){
+
+        if(begin_stopping_downtrack <= current_lanelet_ending_downtrack){
+          // Complete maneuver plan with a stop and wait maneuver if the current lanelet intercepts the stopping downtrack
+          addStopAndWaitToEndOfPlan(resp, downtrack_progress, begin_stopping_downtrack, wm_->getRouteEndTrackPos().downtrack, 
+                          speed_progress, target_speed, stopping_deceleration, current_lanelet_ending_downtrack, current_lanelet, time_progress);
+          return;
+        }
+        else{
+          // First maneuver is lane-following to enable sufficient time to activate turn signals before conducting lane change
+          if(first_maneuver){
+            resp->new_plan.maneuvers.push_back(composeLaneFollowingManeuverMessage(downtrack_progress, current_lanelet_ending_downtrack,  
+                                    speed_progress, target_speed, current_lanelet.id(), time_progress));
+            first_maneuver = false;
+          }
+          else{
+
+            if(!has_planned_upcoming_lc_){
+              // Determine if the ego vehicle's current lanelet is in the rightmost lane
+              bool ego_in_rightmost_lane = false;
+              if(wm_->getMapRoutingGraph()->rights(current_lanelet).size() == 0){
```

Comment:
does checking the size of right lanelets work? I thought the output of 'rights' is boost optional!

Reply:
The `rights()` function definition in its header file ([Click Here](https://github.com/usdot-fhwa-stol/autoware.ai/blob/eb0fcb946c62e094c24d5c71f4059da77f7e0fc1/lanelet2/lanelet2_routing/include/lanelet2_routing/RoutingGraph.h#L284)) shows that it returns a `ConstLanelets` object, which is empty if no routable lanelets to the right are found. For functions in that file that return optionals, the function definition comments state that in certain cases "Nothing" can be returned.

Reply:
sounds good. I was just checking